dali/kernels/signal/moving_mean_square_gpu.cu

```diff
@@ -0,0 +1,273 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cuda_runtime.h>
+#include <cstdio>
+#include <iostream>
+#include <numeric>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/core/util.h"
+#include "dali/kernels/signal/moving_mean_square.h"
+#include "dali/kernels/signal/moving_mean_square_gpu.h"
+
+#define SHARED_MEMORY_BANKS 32
+#define LOG_MEM_BANKS 5
+#define CONFLICT_FREE_OFFSET(n) ((n) >> LOG_MEM_BANKS)
+
+namespace dali {
+namespace kernels {
+namespace signal {
+
+/**
+ * @brief Sample descriptor
+ */
+struct SampleDesc {
+  void *out;
+  const void *in;
+  int64_t len;
+};
```

Comment:
Hmm.... why isn't it a template? I think we had some issues with `half` before it was finally properly handled; now it should work.